index-fetch.js

```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+const Agent = require('./lib/agent')
+
+const globalDispatcher = new Agent()
+
+let fetchImpl = null
+module.exports.fetch = async function fetch (resource, init) {
+  if (!fetchImpl) {
+    fetchImpl = require('./lib/fetch')
+  }
+  return fetchImpl.call(globalDispatcher, resource, init)
+}
```

Comment:
Is it necessary to lazy-load `fetchImpl` ?

Also, the small change I did in https://github.com/nodejs/undici/commit/3e267ece5f3a6b9b4cd6562f885a6884e3926937#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346 should be ported.